src/muya/lib/contentState/updateCtrl.js

```diff
@@ -17,14 +17,66 @@ const INLINE_UPDATE_FRAGMENTS = [
 const INLINE_UPDATE_REG = new RegExp(INLINE_UPDATE_FRAGMENTS.join('|'), 'i')
 
 const updateCtrl = ContentState => {
-  // handle task list item checkbox click
-  ContentState.prototype.listItemCheckBoxClick = function (checkbox) {
-    const { checked, id } = checkbox
-    const block = this.getBlock(id)
+  ContentState.prototype.setCheckBoxState = function (checkbox, checked) {
+    checkbox.checked = checked
+    const block = this.getBlock(checkbox.id)
     block.checked = checked
     checkbox.classList.toggle(CLASS_OR_ID.AG_CHECKBOX_CHECKED)
   }
 
+  ContentState.prototype.getParentCheckBox = function (checkbox) {
+    const parent = checkbox.parentElement.parentElement.parentElement
+    if (parent.id !== CLASS_OR_ID.AG_EDITOR_ID) {
+      return parent.firstElementChild
```

Comment:
Are you sure this firstElementChild is `checkbox`? we support put task list in other block like `blockquote`.

Reply:
I'm not sure now.